File: samples/java_springboot/54.teams-task-module/src/main/java/com/microsoft/bot/sample/teamstaskmodule/models/TaskModuleResponseFactory.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.microsoft.bot.sample.teamstaskmodule.models;

import com.microsoft.bot.schema.teams.TaskModuleContinueResponse;
import com.microsoft.bot.schema.teams.TaskModuleMessageResponse;
import com.microsoft.bot.schema.teams.TaskModuleResponse;
import com.microsoft.bot.schema.teams.TaskModuleTaskInfo;

public final class TaskModuleResponseFactory {
    public static TaskModuleResponse createResponse(String message) {
        TaskModuleMessageResponse taskModuleMessageResponse = new TaskModuleMessageResponse();
        taskModuleMessageResponse.setValue(message);
        TaskModuleResponse taskModuleResponse = new TaskModuleResponse();
        taskModuleResponse.setTask(taskModuleMessageResponse);
        return taskModuleResponse;
    }

    public static TaskModuleResponse createResponse(TaskModuleTaskInfo taskInfo) {
        TaskModuleContinueResponse taskModuleContinueResponse = new TaskModuleContinueResponse();
        taskModuleContinueResponse.setValue(taskInfo);
        TaskModuleResponse taskModuleResponse = new TaskModuleResponse();
        taskModuleResponse.setTask(taskModuleContinueResponse);
        return taskModuleResponse;
    }

    public static TaskModuleResponse toTaskModuleResponse(TaskModuleTaskInfo taskInfo) {
        return createResponse(taskInfo);
    }
}

File: samples/csharp_dotnetcore/54.teams-task-module/Models/TaskModuleResponseFactory.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using Microsoft.Bot.Schema.Teams;

namespace Microsoft.BotBuilderSamples.Models
{
    public static class TaskModuleResponseFactory
    {
        public static TaskModuleResponse CreateResponse(string message)
        {
            return new TaskModuleResponse
            {
                Task = new TaskModuleMessageResponse()
                {
                    Value = message,
                },
            };
        }

        public static TaskModuleResponse CreateResponse(TaskModuleTaskInfo taskInfo)
        {
            return new TaskModuleResponse
            {
                Task = new TaskModuleContinueResponse()
                {
                    Value = taskInfo,
                },
            };
        }

        public static TaskModuleResponse ToTaskModuleResponse(this TaskModuleTaskInfo taskInfo)
        {
            return CreateResponse(taskInfo);
        }
    }
}
